mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_token.go

```diff
@@ -0,0 +1,233 @@
+package resourcemanager
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-provider-google/google/fwtransport"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	"google.golang.org/api/iamcredentials/v1"
+)
+
+var _ ephemeral.EphemeralResource = &googleEphemeralServiceAccountAccessToken{}
+
+func GoogleEphemeralServiceAccountAccessToken() ephemeral.EphemeralResource {
+	return &googleEphemeralServiceAccountAccessToken{}
+}
+
+type googleEphemeralServiceAccountAccessToken struct {
+	providerConfig *fwtransport.FrameworkProviderConfig
+}
+
+func (p *googleEphemeralServiceAccountAccessToken) Metadata(ctx context.Context, req ephemeral.MetadataRequest, resp *ephemeral.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_account_token"
+}
+
+type ephemeralServiceAccountAccessTokenModel struct {
+	TargetServiceAccount types.String `tfsdk:"target_service_account"`
+	AccessToken          types.String `tfsdk:"access_token"`
+	Scopes               types.Set    `tfsdk:"scopes"`
+	Delegates            types.Set    `tfsdk:"delegates"`
+	Lifetime             types.String `tfsdk:"lifetime"`
+}
+
+func (p *googleEphemeralServiceAccountAccessToken) Schema(ctx context.Context, req ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"target_service_account": schema.StringAttribute{
+				Required: true,
+				Validators: []validator.String{
+					serviceAccountNameValidator{},
+				},
+			},
+			"access_token": schema.StringAttribute{
+				Sensitive: true,
+				Computed:  true,
+			},
+			"lifetime": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Validators: []validator.String{
+					durationValidator{
+						maxDuration: 3600 * time.Second,
+					}},
+			},
+			"scopes": schema.SetAttribute{
+				Required:    true,
+				ElementType: types.StringType,
+				// ValidateFunc is not yet supported on lists or sets.
```

Comment:
Is this comment relevant in the context of the plugin-framework? Can we add a Validator to scopes?

Reply:
This new validator was added in the latest commit